qitmeer-miner --help & help command not work properly

### --help & -h
The output shows a mass
```
$ ./qitmeer-miner --help
2019-11-23|11:29:03.548 [INFO ] config parse error                  module=miner error="Usage:\n  qitmeer-miner [OPTIONS]\n\nDebug Command:\n  -l, --listdevices    List number of devices.\n  -v, --version        show the version of miner\n\nThe Config File Options:\n  -C, --configfile=    Path to configuration file\n      --minerlog=      Write miner log file\n\nThe Necessary Config Options:\n  -P, --pow=           blake2bd|cuckaroo|cuckatoo (cuckaroo)\n  -S, --symbol=        Symbol (PMEER)\n  -N, --network=       network privnet|testnet|mainnet (testnet)\n\nThe Solo Config Option:\n  -M, --mineraddress=  Miner Address\n  -s, --rpcserver=     RPC server to connect to (127.0.0.1)\n  -u, --rpcuser=       RPC username\n  -p, --rpcpass=       RPC password\n      --randstr=       Rand String,Your Unique Marking. (Come from Qitmeer!)\n      --notls          Do not verify tls certificates (true)\n      --rpccert=       RPC server certificate chain for validation\n\nThe pool Config Option:\n  -o, --pool=          Pool to connect to (e.g.stratum+tcp://pool:port)\n  -m, --pooluser=      Pool username\n  -n, --poolpass=      Pool password\n\nThe Optional Config Option:\n      --cpuminer       CPUMiner (false)\n      --proxy=         Connect via SOCKS5 proxy (eg. 127.0.0.1:9050)\n      --proxyuser=     Username for proxy server\n      --proxypass=     Password for proxy server\n      --trimmerTimes=  the cuckaroo trimmer times (15)\n      --intensity=     Intensities (the work size is 2^intensity) per device. Single global value or a comma separated list. (24)\n      --worksize=      The explicitly declared sizes of the work to do per device (overrides intensity). Single global value or a comma separated list. (256)\n      --timeout=       rpc timeout. (60)\n      --use_devices=   all gpu devices,you can use ./qitmeer-miner -l to see. examples:0,1 use the #0 device and #1 device\n      --max_tx_count=  max pack tx count (1000)\n      --max_sig_count= max sign tx count (4000)\n      --log_level=     info|debug|error|warn|trace (info)\n      --stats_server=  stats web server (127.0.0.1:1235)\n      --edge_bits=     edge bits (24)\n      --local_size=    local size (4096)\n      --group_size=    work group size (256)\n\nHelp Options:\n  -h, --help           Show this help message\n"
```
```
$ ./qitmeer-miner -h
2019-11-23|11:33:48.568 [INFO ] config parse error                  module=miner error="Usage:\n  qitmeer-miner [OPTIONS]\n\nDebug Command:\n  -l, --listdevices    List number of devices.\n  -v, --version        show the version of miner\n\nThe Config File Options:\n  -C, --configfile=    Path to configuration file\n      --minerlog=      Write miner log file\n\nThe Necessary Config Options:\n  -P, --pow=           blake2bd|cuckaroo|cuckatoo (cuckaroo)\n  -S, --symbol=        Symbol (PMEER)\n  -N, --network=       network privnet|testnet|mainnet (testnet)\n\nThe Solo Config Option:\n  -M, --mineraddress=  Miner Address\n  -s, --rpcserver=     RPC server to connect to (127.0.0.1)\n  -u, --rpcuser=       RPC username\n  -p, --rpcpass=       RPC password\n      --randstr=       Rand String,Your Unique Marking. (Come from Qitmeer!)\n      --notls          Do not verify tls certificates (true)\n      --rpccert=       RPC server certificate chain for validation\n\nThe pool Config Option:\n  -o, --pool=          Pool to connect to (e.g.stratum+tcp://pool:port)\n  -m, --pooluser=      Pool username\n  -n, --poolpass=      Pool password\n\nThe Optional Config Option:\n      --cpuminer       CPUMiner (false)\n      --proxy=         Connect via SOCKS5 proxy (eg. 127.0.0.1:9050)\n      --proxyuser=     Username for proxy server\n      --proxypass=     Password for proxy server\n      --trimmerTimes=  the cuckaroo trimmer times (15)\n      --intensity=     Intensities (the work size is 2^intensity) per device. Single global value or a comma separated list. (24)\n      --worksize=      The explicitly declared sizes of the work to do per device (overrides intensity). Single global value or a comma separated list. (256)\n      --timeout=       rpc timeout. (60)\n      --use_devices=   all gpu devices,you can use ./qitmeer-miner -l to see. examples:0,1 use the #0 device and #1 device\n      --max_tx_count=  max pack tx count (1000)\n      --max_sig_count= max sign tx count (4000)\n      --log_level=     info|debug|error|warn|trace (info)\n      --stats_server=  stats web server (127.0.0.1:1235)\n      --edge_bits=     edge bits (24)\n      --local_size=    local size (4096)\n      --group_size=    work group size (256)\n\nHelp Options:\n  -h, --help           Show this help message\n"
```
### help

The error msg does not show consistently  
```
2019-11-23|11:30:08.151 [WARN ] Don't have config file.             module=miner
2019-11-23|11:30:08.151 [ERROR] Solo need address -M , pool need -o pool address 
```

```
$ ./qitmeer-miner help
2019-11-23|11:30:08.151 [WARN ] Don't have config file.             module=miner
2019-11-23|11:30:08.151 [ERROR] Solo need address -M , pool need -o pool address module=miner
Usage:
  qitmeer-miner [OPTIONS]

Debug Command:
  -l, --listdevices    List number of devices.
  -v, --version        show the version of miner

The Config File Options:
  -C, --configfile=    Path to configuration file
      --minerlog=      Write miner log file

The Necessary Config Options:
  -P, --pow=           blake2bd|cuckaroo|cuckatoo (cuckaroo)
  -S, --symbol=        Symbol (PMEER)
  -N, --network=       network privnet|testnet|mainnet (testnet)

The Solo Config Option:
  -M, --mineraddress=  Miner Address
  -s, --rpcserver=     RPC server to connect to (127.0.0.1)
  -u, --rpcuser=       RPC username
  -p, --rpcpass=       RPC password
      --randstr=       Rand String,Your Unique Marking. (Come from Qitmeer!)
      --notls          Do not verify tls certificates (true)
      --rpccert=       RPC server certificate chain for validation

The pool Config Option:
  -o, --pool=          Pool to connect to (e.g.stratum+tcp://pool:port)
  -m, --pooluser=      Pool username
  -n, --poolpass=      Pool password

The Optional Config Option:
      --cpuminer       CPUMiner (false)
      --proxy=         Connect via SOCKS5 proxy (eg. 127.0.0.1:9050)
      --proxyuser=     Username for proxy server
      --proxypass=     Password for proxy server
      --trimmerTimes=  the cuckaroo trimmer times (15)
      --intensity=     Intensities (the work size is 2^intensity) per device. Single global value or a comma separated list. (24)
      --worksize=      The explicitly declared sizes of the work to do per device (overrides intensity). Single global value or a comma separated list. (256)
      --timeout=       rpc timeout. (60)
      --use_devices=   all gpu devices,you can use ./qitmeer-miner -l to see. examples:0,1 use the #0 device and #1 device
      --max_tx_count=  max pack tx count (1000)
      --max_sig_count= max sign tx count (4000)
      --log_level=     info|debug|error|warn|trace (info)
      --stats_server=  stats web server (127.0.0.1:1235)
      --edge_bits=     edge bits (24)
      --local_size=    local size (4096)
      --group_size=    work group size (256)

Help Options:
  -h, --help           Show this help message
```

--- a/common/config.go
+++ b/common/config.go
@@ -245,7 +245,7 @@ func LoadConfig() (*GlobalConfig, []string, error) {
 	}
 	_,err = preParser.Parse()
 	if err != nil{
-		MinerLoger.Info("config parse error","error", err)
+		_,_ = fmt.Fprintln(os.Stderr, err)
 		os.Exit(0)
 	}
 	if deviceCfg.ListDevices{

--- a/common/device.go
+++ b/common/device.go
@@ -2,6 +2,7 @@ package common
 
 import (
 	"github.com/Qitmeer/go-opencl/cl"
+	`golang.org/x/exp/errors/fmt`
 )
 
 var DevicesTypesForGPUMining = cl.DeviceTypeGPU
@@ -22,9 +23,8 @@ func GetDevices(t cl.DeviceType) []*cl.Device {
 		}
 		for _, device := range platormDevices {
 			clDevices = append(clDevices, device)
-			MinerLoger.Info("Found Device","platform",platform.Name(),"minerID",i,"deviceName",device.Name(),
-				"MaxWorkGroupSize(MB)",device.MaxWorkGroupSize(),"MaxMemAllocSize(MB)",float64(device.MaxMemAllocSize())/1024.00/1024.00 )
-
+			MinerLoger.Info(fmt.Sprintf("Platform:%s DeviceID:# %d DeviceName:%s MaxAllocSize:%.2f MB MaxMemSize:%.2f MB",platform.Name(),i,device.Name(),
+				float64(device.MaxMemAllocSize())/1024.00/1024.00,float64(device.GlobalMemSize())/1024.00/1024.00))
 			i++
 		}
 	}

--- a/symbols/qitmeer/robot.go
+++ b/symbols/qitmeer/robot.go
@@ -95,11 +95,12 @@ func (this *QitmeerRobot)Run() {
 	// Device Miner
 	for _,dev := range this.Devices{
 		dev.SetIsValid(true)
-		dev.InitDevice()
-		dev.SetPool(this.Pool)
 		if len(this.UseDevices) > 0 && !common.InArray(strconv.Itoa(dev.GetMinerId()),this.UseDevices){
 			dev.SetIsValid(false)
+			continue
 		}
+		dev.SetPool(this.Pool)
+		dev.InitDevice()
 		this.Wg.Add(1)
 		go dev.Mine(this.Wg)
 		this.Wg.Add(1)

